daemon/src/main/scala/io/gearpump/cluster/worker/Worker.scala

```diff
@@ -437,18 +439,19 @@ private[cluster] object Worker {
       val format = new java.text.SimpleDateFormat(datePattern)
       format.format(timestamp)
     }
-  }
 
-  private def filterDaemonLib(classPath: Array[String]): Array[String] = {
-    val jarFile = new File(getClass.getProtectionDomain.getCodeSource.getLocation.toURI.getPath)
-    val daemonDir = jarFile.getParent
-    classPath.filterNot{path =>
-      //The class path maybe a symbol link
-      val canonicalized = Try(new File(path).getCanonicalPath)
-      if(canonicalized.isSuccess && canonicalized.get.startsWith(daemonDir)) {
-        true
-      } else {
-        false
+    private def filterDaemonLib(classPath: Array[String]): Array[String] = {
+      val classpathWhiteList = getClasspathWhiteList(config)
+      classPath.filter{ path =>
+        //The class path maybe a symbol link
+        val filePath = Try(new File(path).getCanonicalPath)
+        val parentPath = Try(new File(path).getParentFile.getCanonicalPath)
+        if((filePath.isSuccess && classpathWhiteList.contains(filePath.get)) ||
```

Comment:
What filter here mean?
What filePath.isSuccess mean?
This function is not clear about what it try to do.